activesupport/lib/active_support/core_ext/module/attribute_accessors_per_thread.rb

```diff
@@ -118,6 +118,8 @@ def #{sym}=(obj)
   #   class Customer < Account
   #   end
   #
+  #   Account.user = 'DHH'
```

Comment:
As Matthew said, the value is already set a few lines up and this example tries to show that even though inheritance makes this attribute available to child classes, they don't share the same scope. I'm giving this a close but thanks anyway @mohitnatoo !